src/Server.kt

```diff
@@ -581,8 +606,9 @@ fun SedML.getFskSimulations(parameterMetadata: JsonNode): List<FskSimulation> {
 
 /**
  * Run model with the simulationIndex if provided or the selected simulation in model file (SED-ML).
+ * UPDATE: if a userDefinedSim was given with the POST endpoint, then run that simulation and ignore simulationIndex
  */
-fun FskModel.run(simulationIndex: Int): String {
+fun FskModel.run(simulationIndex: Int, userDefinedSim:FskSimulation?): String {
```

Comment:
This is too confusing in my opinion. We are passing now the simulation with two parameters with one being nullable. I would rather just pass a single non-nullable *simulation: FskSimulation* parameter. Since *run* is only called once we can sort the simulation before the call.

Reply:
done